packages/core/analyzer-import-alias-plugin/src/index.ts

```diff
@@ -34,10 +34,12 @@ export type AppliedTransform = {
 
 export type ImportAliasPluginOptions = {
   [moduleName: string]: {
-    '*'?: (importName: string) => string;
-    override?: {
-      [importName: string]: string;
-    };
+    /**
+     * @privateRemarks
+     * Not convinced we need to enforce catch-all to be a function.
+     * If users want to do a static replacement like `{ '*': 'noop' }` then so be it.
```

Comment:
What use cases are you invisioning that the user would want to do this? In that case, from this library they'd just be basically setting every superclass name to the same thing, which I don't think would ever make sense. Unless the library you're importing only provides one single superclass, but I think that would still be unclear and potentilaly cause a maintainability issue for the user.

```
import { MyButton, MyTextBox } from 'my-library';

export class ExampleButton extends MyButton {}
export class ExampleTextBox extends MyTextBox {}
```
Then using `importAliasPlugin({ ['my-library']: { '*': 'X' }});
Both classes would extend the same thing?

Reply:
I was mainly trying to simplify the typings API and leave `{ '*': 'X' }` undocumented, but perhaps we can keep an explicit type for *. It was late ;-)

I thought about lower-level cases like a `base-components` lib, where say everything extends a common baseline, say FASTElement, but perhaps FASTElement is slightly broken from an LSP manifest standpoint and you want to re-map these to something you've patched for the manifest output, without actually changing the actual superclass. Obviously, you can do that using the function.

I also considered some prior interactions I've had with https://webpack.js.org/configuration/resolve/#resolvealias, namely around the table you see there... and thought about the case of silencing or disabling the parser from going any deeper on the inheritance chain for a lib, `{ '*': false }`.

Happy to try to enforce * being a function again if you prefer.

Reply:
> where say everything extends a common baseline
Ok I can see what you mean. I would think that usually a common baseline like that though would be defining under the hood kind of functionality, which you'd not be exposing on a template. The analyzer is mostly concerened about pulling in properties and attributes. 

> and thought about the case of silencing or disabling the parser from going any deeper on the inheritance chain for a lib, { '*': false }.
The analyzer works by only going one inherintance chain deep anyway. If you have a chain like this `x -extends-> y -extends-> z` then when you want to run the analyzer on `X` then you'd only get `Y` by default, to get `Z` you runn to run the analyzer in `Y` first to get all of `Z`.

> Happy to try to enforce * being a function again if you prefer.
I think I would prefer this, if the typescript typing can be made clean. If it isn't then happy to leave it as it is, as long as the instructions don't say about setting it as a string. Someone who is very good and knows what they're doing can achieve it anyway, but don't want to confuse people just getting started

Reply:
Thanks for the extra detail, sure, will restrict catch-all to function types only.

Reply:
Done.